oak-security-spi/src/main/java/org/apache/jackrabbit/oak/spi/security/authentication/AbstractLoginModule.java

```diff
@@ -212,9 +215,8 @@ public void initialize(Subject subject, CallbackHandler callbackHandler, Map<Str
     @Override
     public boolean logout() throws LoginException {
         boolean success = false;
-        Set<Object> creds = ImmutableSet.builder()
-                .addAll(subject.getPublicCredentials(Credentials.class))
-                .addAll(subject.getPublicCredentials(AuthInfo.class)).build();
+        Set<Object> creds = Stream.concat(subject.getPublicCredentials(Credentials.class).stream(), subject.getPublicCredentials(AuthInfo.class).stream())
```

Comment:
Did you check that it's ok not to make this immutable?

Reply:
Yes, it is only iterated in the same class to destroy these credentials, but I would say that we should play safe here as well. let me update the ticket to use `Collections.unmodifiableSet`

Reply:
addressed in https://github.com/apache/jackrabbit-oak/pull/2212/commits/1c2d8a0e87263e3b7618119e901d34ec040cd478